### ISSUE TITLE
Finger, AdvancedThrowing and Wipe Goggles ignore the reload mutex causing animation issues

### ISSUE THREAD

**Issue by rebelvg** [acemod/ACE3#3938]
**Arma 3 Version:** `1.60` (stable)
**CBA Version:** `2.4.1` (stable)
**ACE3 Version:** `3.5.1` (eedb406ae5605c2dd5663fd0b2de2bb9bf501b6b with advancedThrowing branch)

**Mods:**
- `@CBA_A3`
- `@ace`

**Description:**
- If you do any action mid-reloading that will trigger an animation, it will freeze the weapon animation and will not return a mag that animation took from the gun at the start of the reloading. Addons that do not check if unit is reloading so far I found advanced throwing and finger pointing. Weapon switching actually has the same problem as vanilla arma, weapon animation freezes, but the unit animation continues and it returns the mag in the end so it's not a huge deal.
Video.
https://www.youtube.com/watch?v=WwmbtaGV05I
First reload, without interupting.
Second, pointing. Interrupts reload animation, doesn't return the mag reload took at the start.
Third, grenade throw. Same.

**Steps to reproduce:**
- Pick something with a long reload animation.
- Point forward mid-reloading.
- Throw a grenade mid-reloading.

**Where did the issue occur?**
- Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Placed Modules:**
- Pointing Module.

**RPT log file:**
- N/A.

**Comment by esteldunedain** [comment]
What key or keyCombination do you use for reloading? We have a mutex in place so you theoretically shouldn't be able to use gestures while reloading (the actions should be unavailable).

As for the grenade throwing, I'm not sure what's causing it.

BTW, is that a vanilla weapon?

**Comment by commy2** [comment]
This has been around since A3 (e.g. AGM bipods). I'm for changing the tag to "backlog" since there is absolutely nothing _we_ can do.

**Comment by esteldunedain** [comment]
Hold on the gestures thing could be a bug; as I said, they shouldn't be possible (unless he's using a joystick)

**Comment by esteldunedain** [comment]
AS I edited above:  the grenade thing is due to the presence of "advanced throwing". That should take the reloadmutex into account @jonpas.

**Comment by jokoho48** [comment]
i think currently only Gestures use a Reload Mutex and Advanced Throwing and Point dont have a Mutex

**Comment by jonpas** [comment]
I will add reload mutex to advanced throwing.

Changed milestone to 3.7.0 as advanced throwing won't be in 3.6.0.

**Comment by rebelvg** [comment]
I use R key and yes, that's a vanilla weapon. Heavy Gunner for NATO side.
It's the same for all other weapons, more prominent here cause of the long and unusual reloading.
Was afk, but I see you figured everything out.

**Comment by esteldunedain** [comment]
@jokoho48, I was confusing `ace_finger` with `ace_gestures`. So finger should use the reload mutex too.

**Comment by jokoho48** [comment]
finger and Advanced Throwing yeah but for this we need to port the reload mutex in to common

**Comment by esteldunedain** [comment]
> finger and Advanced Throwing yeah but for this we need to port the reload mutex in to common

Yeah, that sounds reasonable.

**Comment by esteldunedain** [comment]
Mm, I'm also thinking about this @commy:

```sqf
["isNotReloadingDead", {
    params ["_unit"];
    !GVAR(isReloading)
}] call FUNC(addCanInteractWithCondition);
```

**Comment by commy2** [comment]
sure. makes sense to me.

**Comment by jonpas** [comment]
We can do this for 3.6.1. @esteldunedain want to implement the system and apply it to finger pointing? I am not entirely sure how all this works that you described above, I'll add it to advanced throwing though.

**Comment by rebelvg** [comment]
Wipe glasses action needs reload mutex too.

**Comment by jonpas** [comment]
Fixed by #4062
